src/api/app/policies/assignment_policy.rb

```diff
@@ -4,10 +4,10 @@
 class AssignmentPolicy < ApplicationPolicy
   def create?
     return false unless Flipper.enabled?(:foster_collaboration, user)
-
     return true if user.admin?
+    return false unless PackagePolicy.new(user, record.package).assign?
 
-    record.assignee_is_a_collaborator?
+    record.assignee_has_required_role_to_be_assigned?
```

Comment:
Why do you run this validation here?

Reply:
It was already like that, I just adapted then naming and logic inside to be correct. Didn't wanted to go further into refactors.